### ISSUE TITLE
no automatic resizing after addChild (mx)

### ISSUE THREAD

**Issue by tranquiliste** [apache/royale-asjs#903]
Hello,

Following Yishay advice, I create this issue about DOM note automatically resising after adding a child.
I have created a test project wich  displays a button to add 4 labels to the dom
By default labels are added but parent DOM size is not modified, so you can view the labels only by scrolling
If you uncomment the line 33 of the Ecran2Class ( fire "layoutNeeded" event) you get the behaviour I would like to have.

I use 0.9.7 to compile

Thanks for your help

Nicolas

**Comment by tranquiliste** [comment]
[testDynamicLayout.zip](https://github.com/apache/royale-asjs/files/5172905/testDynamicLayout.zip)
with the test project attached

**Comment by tranquiliste** [comment]
Hello,

I tried to look at the issue, but could not find  what's wrong. Height of DOM inside which we added the four labels is correctly calculated but not its parent.
Is it a matter of adding a bead? if so I could not find which one.

Any help would be apreciated.
Thanks
Nicolas

**Comment by tranquiliste** [comment]
Hello me again,
After further testing I resalized that if my parent DOM (Ecran2Class)  extends Panel rather than Box, the dispatchEvent( new FlexEvent("layoutNeeded")); has no effect.

Nicolas

**Comment by aharui** [comment]
This appears to be fixed by commit 241c927e5b5ad5a8e7264b2e21072aa1be6fc959 in the ChildResize branch, although there was a bug with Panel that I just fixed with commit f59cbc30fe3e9acc8be59845ed1f18e97139666a

**Comment by tranquiliste** [comment]
Hello ,

Thanks for the information. Have these 2 fixes been integrated in the
nightly build?

Nicolas

**Comment by aharui** [comment]
I don't think so.  Not sure what the plan is for that branch.  You could pull down the commits and build locally and see if it works for you, then propose a PR with those two commits.

**Comment by tranquiliste** [comment]
Hello aharui,

Thanks for the information, I looked at the branches and there are other differences compared to the nighlybuild which seem interresting.
Why no merge has been done?

Concerning a local build I need to see if it is difficult as my environment is not configured for that
Nicolas

**Comment by tranquiliste** [comment]
Hello @aharui I tried your fixes, it works for the addChild, but not for the remove (height is not re-calculated). I imagine this has to do with a childRemoved function, is that right? can you give me some direction for the fix?

Nicolas

**Comment by tranquiliste** [comment]
thanks very much, I am going to test that.

Nicolas
